internal/policy/verify.go

```diff
@@ -747,11 +757,27 @@ func (v *Verifier) Verify(ctx context.Context, gitObjectID gitinterface.Hash, en
 				continue
 			}
 
-			verifier, err := signerverifier.NewSignerVerifierFromTUFKey(key) //nolint:staticcheck
-			if err != nil && !errors.Is(err, common.ErrUnknownKeyType) {
-				return nil, err
+			var (
+				envVerifier sslibdsse.Verifier
+				err         error
+			)
+			switch key.KeyType {
+			case ssh.KeyType:
+				envVerifier, err = ssh.NewVerifierFromKey(key)
+				if err != nil {
+					return nil, err
+				}
+
+				envVerifiers = append(envVerifiers, envVerifier)
+			case gpg.KeyType:
+				slog.Debug(fmt.Sprintf("Found GPG key '%s', cannot use for DSSE signature verification yet...", key.KeyID))
+				continue
+			case sigstore.KeyType:
+				slog.Debug(fmt.Sprintf("Found Sigstore key '%s', cannot use for DSSE signature verification yet...", key.KeyID))
```

Comment:
yeah, I also think that'd be part of adding a key as trusted to the metadata. ie, block adding this gpg key as trusted since root.json says only [ssh, sigstore] are allowed.